aptos-move/block-executor/src/code_cache_global.rs

```diff
@@ -64,88 +54,93 @@ where
             .acquire()
             .get(key)
-            .is_some_and(|module| module.is_valid())
+            .is_some_and(|entry| entry.is_valid())
     }
 
     /// Marks the cached module (if it exists) as invalid. As a result, all subsequent calls to the
-    /// cache for the associated key  will result in a cache miss. Note that it is fine for an
-    /// entry not to exist, in which case this is a no-op.
-    pub(crate) fn mark_invalid(&self, key: &K) {
-        if let Some(module) = self.module_cache.acquire().get(key) {
-            module.mark_invalid();
+    /// cache for the associated key will result in a cache miss. If an entry does not to exist, is
+    /// a no-op.
+    pub fn mark_invalid_if_contains(&self, key: &K) {
+        if let Some(entry) = self.module_cache.acquire().get(key) {
+            entry.mark_invalid();
         }
     }
 
-    /// Returns the module stored in cache. If the module has not been cached, or it exists but is
-    /// not valid, [None] is returned.
-    pub(crate) fn get(&self, key: &K) -> Option<Arc<ModuleCode<DC, VC, E, Option<TxnIndex>>>> {
-        self.module_cache.acquire().get(key).and_then(|module| {
-            if module.is_valid() {
-                Some(module.inner().clone())
-            } else {
-                None
-            }
-        })
+    /// Returns the module stored in cache. If the module has not been cached, or it exists but it
+    /// is not valid, [None] is returned.
+    pub fn get(&self, key: &K) -> Option<Arc<ModuleCode<DC, VC, E>>> {
+        self.module_cache
+            .acquire()
+            .get(key)
+            .and_then(|entry| entry.is_valid().then(|| entry.module_code().clone()))
     }
 
     /// Flushes the cache. Should never be called throughout block-execution. Use with caution.
     pub fn flush_unchecked(&self) {
         self.module_cache.acquire().clear();
+        self.size.store(0, Ordering::Relaxed);
+    }
+
+    /// Returns the number of entries in the cache.
+    pub fn num_modules(&self) -> usize {
+        self.module_cache.acquire().len()
+    }
+
+    /// Returns the sum of serialized sizes of modules stored in cache.
+    pub fn size_in_bytes(&self) -> usize {
+        self.size.load(Ordering::Relaxed)
+    }
+
+    /// Returns true if the sum of serialized sizes of modules stored in cache is greater than the
+    /// specified value.
+    pub fn size_in_bytes_is_greater_than(&self, size: usize) -> bool {
+        self.size_in_bytes() > size
     }
 
     /// Inserts modules into the cache. Should never be called throughout block-execution. Use with
     /// caution.
     ///
     /// Notes:
     ///   1. Only verified modules are inserted.
-    ///   2. Versions of inserted modules are set to [None] (storage version).
-    ///   3. Valid modules should not be removed, and new modules should have unique ownership. If
+    ///   2. Valid modules should not be removed, and new modules should have unique ownership. If
```

Comment:
supernit / suggestion: I feel like "valid module" might be misinterpreted, what valid really is, is the opposite of the classic dirty flag, so we could use "clean", but nobody does that, or dirty (and have default not dirty) - that's standard.

Then here we can say only modules marked as dirty should be removed.

Reply:
My intuition is that if republished, it is not dirty because the content is not modified, but it is invalidated by other write. How about `invalidated` or `overriden`?

Reply:
i've seen dirty used liberally but that's a very good point, overridden sounds perfect with this in mind!

Reply:
Let's rename in the next PR